### ISSUE TITLE
WordCount example fail when execute from external project.

### ISSUE THREAD

**Issue by kikicarbonell** [spark-jobserver/spark-jobserver#241]
My project has:

**build.sbt**
```
name := "demo-spark-app"

version := "1.0"

scalaVersion := "2.11.7"

resolvers += "Job Server Bintray" at "https://dl.bintray.com/spark-jobserver/maven"

libraryDependencies ++= Seq (
  "spark.jobserver" %% "job-server-api" % "0.5.2",
  "org.apache.spark" %% "spark-core" % "1.4.1"
)
```
**src/main/scala/spark/jobserver/WordCountExample**
```
package spark.jobserver

import com.typesafe.config.{Config, ConfigFactory}
import org.apache.spark._
import org.apache.spark.SparkContext._
import scala.util.Try

/**
* A super-simple Spark job example that implements the SparkJob trait and can be submitted to the job server.
*
* Set the config with the sentence to split or count:
* input.string = "adsfasdf asdkf  safksf a sdfa"
*
* validate() returns SparkJobInvalid if there is no input.string
*/
object WordCountExample extends SparkJob {
  def main(args: Array[String]) {
    val sc = new SparkContext("local[4]", "WordCountExample")
    val config = ConfigFactory.parseString("")
    val results = runJob(sc, config)
    println("Result is " + results)
  }

  override def validate(sc: SparkContext, config: Config): SparkJobValidation = {
    Try(config.getString("input.string"))
      .map(x => SparkJobValid)
      .getOrElse(SparkJobInvalid("No input.string config param"))
  }

  override def runJob(sc: SparkContext, config: Config): Any = {
    val dd = sc.parallelize(config.getString("input.string").split(" ").toSeq)
    dd.map((_, 1)).reduceByKey(_ + _).collect().toMap
  }
}
```
After succesfull `sbt package` I submmit jar to job-server as test but when execute command:
```
curl -d "input.string = a b c a b see" 'localhost:8090/jobs?appName=test&classPath=spark.jobserver.WordCountExample'
```
The job-server output show some errors:
```
job-server[ERROR] Exception in thread "pool-19-thread-1" java.lang.NoSuchMethodError: scala.Predef$.$conforms()Lscala/Predef$$less$colon$less;
job-server[ERROR]       at spark.jobserver.WordCountExample$.runJob(WordCountExample.scala:32)
job-server[ERROR]       at spark.jobserver.WordCountExample$.runJob(WordCountExample.scala:16)
job-server[ERROR]       at spark.jobserver.JobManagerActor$$anonfun$spark$jobserver$JobManagerActor$$getJobFuture$4.apply(JobManagerActor.scala:254)
job-server[ERROR]       at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
job-server[ERROR]       at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
job-server[ERROR]       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
job-server[ERROR]       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
job-server[ERROR]       at java.lang.Thread.run(Thread.java:745)

```
I don't understand the problem because previously I tried the all class of job-server-test from job-server source code and all work fine. The other examples of job-server-test work fine on my project, the only problem is with Word Count.

**Comment by velvia** [comment]
Did you package your spark jobserver with Scala 2.10?   Your example is
Scala 2.11, so you either need to compile your jar with 2.10 or compile SJS
with 2.11.... your choice.

On Wed, Sep 2, 2015 at 6:56 AM, Enrique Carbonell <notifications@github.com>
wrote:

> My project has:
>
> *build.sbt*
>
> name := "demo-spark-app"
>
> version := "1.0"
>
> scalaVersion := "2.11.7"
>
> resolvers += "Job Server Bintray" at "https://dl.bintray.com/spark-jobserver/maven"
>
> libraryDependencies ++= Seq (
>   "spark.jobserver" %% "job-server-api" % "0.5.2",
>   "org.apache.spark" %% "spark-core" % "1.4.1"
> )
>
> *src/main/scala/spark/jobserver/WordCountExample*
>
> package spark.jobserver
>
> import com.typesafe.config.{Config, ConfigFactory}
> import org.apache.spark._
> import org.apache.spark.SparkContext._
> import scala.util.Try
>
> /**
> * A super-simple Spark job example that implements the SparkJob trait and can be submitted to the job server.
> *
> * Set the config with the sentence to split or count:
> * input.string = "adsfasdf asdkf  safksf a sdfa"
> *
> * validate() returns SparkJobInvalid if there is no input.string
> */
> object WordCountExample extends SparkJob {
>   def main(args: Array[String]) {
>     val sc = new SparkContext("local[4]", "WordCountExample")
>     val config = ConfigFactory.parseString("")
>     val results = runJob(sc, config)
>     println("Result is " + results)
>   }
>
>   override def validate(sc: SparkContext, config: Config): SparkJobValidation = {
>     Try(config.getString("input.string"))
>       .map(x => SparkJobValid)
>       .getOrElse(SparkJobInvalid("No input.string config param"))
>   }
>
>   override def runJob(sc: SparkContext, config: Config): Any = {
>     val dd = sc.parallelize(config.getString("input.string").split(" ").toSeq)
>     dd.map((_, 1)).reduceByKey(_ + _).collect().toMap
>   }
> }
>
> After succesfull sbt package I submmit jar to job-server as test but when
> execute command:
>
> curl -d "input.string = a b c a b see" 'localhost:8090/jobs?appName=test&classPath=spark.jobserver.WordCountExample'
>
> The job-server output show some errors:
>
>  job-server[ERROR] Exception in thread "pool-19-thread-1" java.lang.NoSuchMethodError: scala.Predef$.$conforms()Lscala/Predef$$less$colon$less;
> job-server[ERROR]       at spark.jobserver.WordCountExample$.runJob(WordCountExample.scala:32)
> job-server[ERROR]       at spark.jobserver.WordCountExample$.runJob(WordCountExample.scala:16)
> job-server[ERROR]       at spark.jobserver.JobManagerActor$$anonfun$spark$jobserver$JobManagerActor$$getJobFuture$4.apply(JobManagerActor.scala:254)
> job-server[ERROR]       at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
> job-server[ERROR]       at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
> job-server[ERROR]       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
> job-server[ERROR]       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
> job-server[ERROR]       at java.lang.Thread.run(Thread.java:745)
>
>
> I don't understand the problem because previously I tried the all class of
> job-server-test from job-server source code and all work fine. The other
> examples of job-server-test work fine on my project, the only problem is
> with Word Count.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spark-jobserver/spark-jobserver/issues/241>.
>



-- 
If you are free, you need to free somebody else.
If you have some power, then your job is to empower somebody else.
   --- Toni Morrison

**Comment by kikicarbonell** [comment]
@velvia you are right, I change ScalaVersion to 2.10.4 and all work fine now. But I'm packaging spark-jobserver from tag v0.5.2 and I was using on my project ScalaVersion = 2.11 because the Readme indicate support for Scala 2.10 and 2.11, please verify it. Thanks for your help and patience.

**Comment by velvia** [comment]
Enrique, yes SJS does support Scala 2.11, that's why you were able to
compile your project.  Just need to make sure to compile SJS for the right
version and also use Spark for 2.11 as well.

On Wed, Sep 2, 2015 at 11:00 AM, Enrique Carbonell <notifications@github.com
> wrote:

> Closed #241
> <https://github.com/spark-jobserver/spark-jobserver/issues/241>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spark-jobserver/spark-jobserver/issues/241#event-399249128>
> .
>



-- 
If you are free, you need to free somebody else.
If you have some power, then your job is to empower somebody else.
   --- Toni Morrison
